### ISSUE TITLE
GET Request parameter object won't work

### ISSUE THREAD

**Issue by richardshergold** [sneas/ionic-native-http-connection-backend#6]
FYI

With your module included I'm unable to make any HTTP Get Requests where the parameters are added as an object like this:

this.http
.get(url, { params: {"parm1":140,"parm2":100 })

I do this throughout my app.

To get the app to work I had to modify all my calls so that the parameters are concatenated to the URL string manually and not passed in as a parameter to the get call.

**Comment by sneas** [comment]
Thanks for the feedback, @richardshergold. I'll have a look at it.

**Comment by sneas** [comment]
@richardshergold can you try reproduce the error with https://httpbin.org service ?

e.g.

```typescript
this.http.get('https://httpbin.org/get',  { params: {'parm1': 140, 'parm2': 100 }})
    .subscribe(response => {
        console.log('Response data args', response.json().args); // outputs 'Response data args – {parm1: "140", parm2: "100"}'
    });
```

I've just tried the code above but I received the expected result. It would be very helpful if you find the way to break the library using https://httpbin.org service and share the code snippet. Thanks.

**Comment by richardshergold** [comment]
Hmm, that's odd.  If that works for you then I'm sure it will for me.  The only slightly odd thing I have going on is the name of one of my option parameters contains square brackets i.e my options object looks like this:


`{"sid":"55861f93e494fdab9ec3c7c49c9e4ce84e15542f::feed::100001350","max":20,"offset":0,"filters[]":"Organisation::Sms"}`

The other difference is that in the call I reference this as an object name i.e

```
let options = {"sid":"55861f93e494fdab9ec3c7c49c9e4ce84e15542f::feed::100001350","max":20,"offset":0,"filters[]":"Organisation::Sms"};

this.http
.get(url, { params: options)
```


Do you think that could be throwing it?  I suppose the other difference (small one) is that my parameters are strings and / or numbers - your example has just numbers?

**Comment by richardshergold** [comment]
Oh - did you perform your test on a device?  The code seems to run fine in a browser (Chrome) it's only on a physical device (iOS) it fails.

**Comment by sneas** [comment]
This might be a silly question but did you install Cordova-HTTP plugin as readme says?
`ionic cordova plugin add https://github.com/sneas/cordova-HTTP`

and if you did, I'm wondering, does this code give you any error?

```
this.http.get('https://httpbin.org/get',  { params:
    {'sid':'55861f93e494fdab9ec3c7c49c9e4ce84e15542f::feed::100001350','max': 20,
        'offset': 0, 'filters[]': 'Organisation::Sms'}})
    .subscribe(response => {
        console.log('Response data args', response.json().args);
    });
```

There is no error in case you see something like this in Safari console:

![image](https://user-images.githubusercontent.com/144651/29423544-754492fc-837c-11e7-81d7-22abcfa40c2d.png)

or something like this in xcode console:

![image](https://user-images.githubusercontent.com/144651/29423585-98db5340-837c-11e7-937e-d4875dba4f1d.png)

**Comment by sneas** [comment]
I've just tried the code above on device and it works as expected.

**Comment by richardshergold** [comment]
Yes I have the plugin installed.

I don't really understand what's happening but if I change my code FROM this:

getFeedItems(url: string, options: any): Observable<any> {
    return this.http
      .get(url,{params:options})
      .timeout(API_GLOBAL.timeout)
      .map((res: Response) => {
        return res.json();
      })
      .catch(this.errorHandler.catchError);
  }

TO this:

getFeedItems(url: string, options: any): Observable<any> {
    url = url + "?sid=" + options.sid;
    url = url + "&max=" + options.max;
    url = url + "&offset=" + options.offset;
    url = url + "&filters[]=" + options["filters[]"];
    return this.http
      .get(url)
      .timeout(API_GLOBAL.timeout)
      .map((res: Response) => {
        return res.json();
      })
      .catch(this.errorHandler.catchError);
  }

then it works!   Without that change, it doesn't.

**Comment by sneas** [comment]
Hi @richardshergold. The problem was with `filters[]` param. It's being encoded by Http in case of passing as `params` object and not encoded in case of passing as a part of URL string. I created bug fix (v1.2.5). Please have a try.

**Comment by sneas** [comment]
Closing the issue. Feel free to reopen it in case of any problem with get params.
